src/main/java/mate/academy/spring/validation/PasswordValidator.java

```diff
@@ -0,0 +1,20 @@
+package mate.academy.spring.validation;
+
+import javax.validation.ConstraintValidator;
+import javax.validation.ConstraintValidatorContext;
+import mate.academy.spring.dto.request.UserRequestDto;
+
+public class PasswordValidator implements ConstraintValidator<Password, UserRequestDto> {
+    private static final String PASSWORD_PATTERN = "(?=^.{8,}$)((?=.*\\d)|(?=.*\\W+))"
+            + "(?![.\\n])(?=.*[A-Z])(?=.*[a-z]).*$";
```

Comment:
When you use such pattern it is a good practice to add a comment to describe its rules. Noone can't understand the requirements when see it.